src/components/PortalButtonSet/PortalButtonSet.js

```diff
@@ -0,0 +1,25 @@
+import React, { useState } from 'react';
+import './PortalButtonSet.css';
+
+function PortalButtonSet({labels, activeButton = 0, onClick}) {
+
+    const [active, setActive] = useState(activeButton);
```

Comment:
It is not the "react way" of initializing a state with a value from a prop.

The fix should be that you don't have a state for this. The comp sends the active index to the parent and it should return it as a prop and cause a rerender.